src/omotes_orchestrator/main.py

```diff
@@ -186,6 +189,12 @@ def start(self) -> None:
         self.postgresql_if.start()
         self._resume_init_barriers(self.postgresql_if.get_all_jobs())
 
+        """Start a thread to run database stale jobs cleaner."""
+        stale_jobs_cleaner = threading.Thread(target=self.postgresql_if.start_stale_jobs_cleaner,
```

Comment:
@lfse-slafleur I am curious to hear what you think regarding the current implementation. I didn't create an isolated component as you described in #38, but instead implemented a function `start_stale_jobs_cleaner()` which starts as a background thread when the orchestrator is up.

If you think the direction still aligns, I will proceed to write a test.

Reply:
Hea Cheng-Kai!

I would prefer to move the function from the PostgresInterface to a separate class & file. Here is my rational: In your current approach we are mixing components (interface to postgres & cleaning the database) in a single class. Also, the Orchestrator class now instantiates and starts the thread while the Orchestrator component should not contain the innerworkings of another component. While the difference is small, it will ensure that all code, with regards to cleaning the database, is isolated from other components which I expect will help with long-term maintainability.

Do you agree or do you have another perspective that might fit better here? Curious to your thoughts!

Reply:
Hey Sebastiaan, 

Thanks for the feedback! I see your points which I like and I agree with you. I will refactor and move this part into a separate class and start the database cleaning thread when the component/instance is initialized.

Reply:
@lfse-slafleur I tried to follow the existing pattern in the codebase and refactored the part accordingly. Again I am curious to hear if there is any thought on this.

Reply:
Good to have you back! Thanks for the review, they are quite helpful and structured inputs. Will be working on them soon.